pulp_rpm/tests/functional/api/test_publish.py

```diff
@@ -333,3 +342,141 @@ def _get_updateinfo_xml_path(root_elem):
         ]
         xpath = '{{{}}}location'.format(RPM_NAMESPACES['metadata/repo'])
         return data_elems[0].find(xpath).get('href')
+
+
+class ChecksumTypeTestCase(unittest.TestCase):
+    """Publish repository and validate the updateinfo.
+
+    This Test does the following:
+
+    1. Create a rpm repo and a remote.
+    2. Sync the repo with the remote.
+    3. Publish and distribute the repo.
+    4. Verify the checksum types
+
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """Create class-wide variables."""
+        delete_orphans()
+        cls.cfg = config.get_config()
+        cls.client = gen_rpm_client()
+        cls.repo_api = RepositoriesRpmApi(cls.client)
+        cls.remote_api = RemotesRpmApi(cls.client)
+        cls.publications = PublicationsRpmApi(cls.client)
+        cls.distributions = DistributionsRpmApi(cls.client)
+
+    def get_checksum_types(self, **kwargs):
+        """Sync and publish an RPM repository."""
+        package_checksum_type = kwargs.get("package_checksum_type")
+        metadata_checksum_type = kwargs.get("metadata_checksum_type")
+        policy = kwargs.get("policy", "immediate")
+        # 1. create repo and remote
+        repo = self.repo_api.create(gen_repo())
+        self.addCleanup(self.repo_api.delete, repo.pulp_href)
+
+        body = gen_rpm_remote(policy=policy)
+        remote = self.remote_api.create(body)
+        self.addCleanup(self.remote_api.delete, remote.pulp_href)
+
+        # 2. Sync it
+        repository_sync_data = RpmRepositorySyncURL(remote=remote.pulp_href)
+        sync_response = self.repo_api.sync(repo.pulp_href, repository_sync_data)
+        monitor_task(sync_response.task)
+
+        # 3. Publish and distribute
+        publish_data = RpmRpmPublication(repository=repo.pulp_href)
```

Comment:
why are you setting the data like this? RpmRpmPublication(repository=repo.pulp_href, package_checksum_type = , metadata_checksum_type = )

Reply:
no reason, I made some changes on the code and it is not necessary anymore, but I forgot to change it